src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+
+        Sub New()
+        End Sub
+
+        Private Function CreateTestFile(TestBase As FileIOTestBase, ByVal TestData() As Char, Optional TestFileNameWithPath As String = "") As String
+            If TestFileNameWithPath.Length = 0 Then
+                TestFileNameWithPath = TestBase.GetTestFilePath()
+            Else
+                Assert.True(IO.Path.IsPathRooted(TestFileNameWithPath))
+                Assert.True(TestFileNameWithPath.StartsWith(TestBase.TestDirectory))
+            End If
+            ' Write and copy file
+            Using sourceStream As New IO.StreamWriter(IO.File.Create(TestFileNameWithPath))
+                sourceStream.Write(TestData, 0, TestData.Length)
+            End Using
+
+            Return TestFileNameWithPath
+        End Function
+
+        <Fact>
+        Public Sub CombinePathTest_BadBaseDirectory_RelativePath()
+            Assert.Throws(Of ArgumentNullException)(Function() FileSystem.CombinePath("", "Test2"))
+            Assert.Equal(FileSystem.CombinePath("C:\", ""), "C:\")
+        End Sub
+
+        <Fact>
+        Public Sub CombinePathTest_BaseDirectory_RelativePath()
+            Assert.Equal(FileSystem.CombinePath("C:\", "Test2"), IO.Path.Combine("C:\", "Test2"))
+        End Sub
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+            Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+            IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+            For i As Integer = 0 To 5
+                CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToSourceDirectory, $"NewFile{i}"))
+            Next
+            FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory)
+            Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                ' Ensure copy transferred written data
+                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
+                    Dim ReadData(SourceData.Length - 1) As Char
+                    stream.Read(ReadData, 0, SourceData.Length)
+                    Assert.Equal(SourceData, ReadData)
+                End Using
+            Next
+            IO.Directory.Delete(FullPathToTargetDirectory, recursive:=True)
+            IO.Directory.CreateDirectory(FullPathToTargetDirectory)
+            CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToTargetDirectory, $"NewFile0"))
+            Assert.Throws(Of IO.IOException)(Sub() FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory))
+            TestBase.Dispose()
```

Comment:
use using statement instead

Reply:
I don't understand what you are asking to be changed? I have only used Using when reading or writing a file.

Reply:
we can use using statement for TestBase variable too